python/pyspark/worker.py

```diff
@@ -178,7 +178,7 @@ def process():
     except Exception:
         try:
             write_int(SpecialLengths.PYTHON_EXCEPTION_THROWN, outfile)
-            write_with_length(traceback.format_exc().encode("utf-8"), outfile)
+            write_with_length(traceback.format_exc(), outfile)
```

Comment:
It slove problem in Python 2 but cause blocked in Python 3.
I have new pr in [#18262](https://github.com/apache/spark/pull/18262) used try except.
I will close this pr.